### PR TITLE
🖕 Transfer emoji from one instance to another

### PR DESCRIPTION
**Pull request by nifengle**
- Adds an emoji transfer service for transfering emoji from one instance to another
- Adds an image client for fetching emoji images from Slack
- Renames emoji_client to slack_emoji_client
- Updates client to accept an API token as init arg
- Update README
- Add seed json file to gitignore

### CODE REVIEW

File: application.py [hunk 1]
```diff
@@ -1,9 +1,11 @@
+# import json
```
Comment (nifengle) [inline application.py]: leaving these comments to make it easier to test locally using this file from the command line

File: .gitignore [hunk 2]
```diff
@@ -106,3 +106,6 @@ venv.bak/
 # editors
 .idea/
 .vscode/
+
+# Seed json
+results.json
```
Comment (clanam) [inline .gitignore]: let's just check it in for now. that way all developers can get easy access to it.

File: README.md [hunk 3]
```diff
@@ -26,9 +26,20 @@ A way to move emoji from one Slack instance to another
 - run `python3 application.py [message]` to send a message to the sandbox.
 
 
-To use the emoji client:
+## Testing in a python console
```
Comment (nifengle) [inline README.md]: Hoping this section heading makes it clearer

File: emoji/emoji_transfer_service.py [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+    emoji_path = source_dict.get(emoji_name)
+
+    if emoji_path.startswith("alias:"):
+        """If it's an alias, we want to get the url for the file"""
+        aliased_from = emoji_path.replace("alias:", "")
+        emoji_path = source_dict.get(aliased_from)
+
+    if emoji_path is None:
+        """Making an alias for a standard emoji"""
+        return destination_client.add_alias(emoji_name, aliased_from)
```
Comment (clanam) [inline emoji/emoji_transfer_service.py]: i don't get the flow here (don't know python). it reads like the alias (above) is looking for the url to copy it.

...but then how does aliased_from get set if emoji_path is empty?

File: emoji/emoji_transfer_service.py [hunk 5]
```diff
@@ -0,0 +1,20 @@
+from emoji import image_client
+
+def transfer(source_client, destination_client, emoji_name):
+    source_dict = source_client.emoji_dict()
+    destination_dict = destination_client.emoji_dict()
+
+    emoji_path = source_dict.get(emoji_name)
+
+    if emoji_path.startswith("alias:"):
+        """If it's an alias, we want to get the url for the file"""
```
Comment (clanam) [inline emoji/emoji_transfer_service.py]: don't we really want to upload the original file first (if white-listed/existing on slack) and then make an alias to it?

File: emoji/emoji_transfer_service.py (hunk 4 above)
Comment (nifengle) [reply emoji/emoji_transfer_service.py]: so the dict looks like
```
{ "emoji-name": "alias:original-emoji" }
```

if `original-emoji` isn't in the dict somewhere (i.e. it's one of the non-custom ones), then `emoji_path` is `None` but `emoji_name` is still `original-emoji`

File: emoji/emoji_transfer_service.py (hunk 5 above)
Comment (nifengle) [reply emoji/emoji_transfer_service.py]: Not really. If our list determines that everyone refers to emoji `foo` as `bar`, we want to upload it as `bar`

Comment (gmkohler) [reply emoji/emoji_transfer_service.py]: Regarding the original file, I think we can iterate on this to collect aliases.  e.g. if I have

```py
destionation_dict = {
  'sad_cowboy': 'http://my-url.com',
  'emo_cowboy': 'alias:sad_cowboy'
}
```
we can read this input:

```py
["emo_cowboy", "sad_cowboy"]
```

and eventually call:
```py
image = image_client.get(destination_dict.get('sad_cowboy'))

destination_client.add_emoji('sad_cowboy', image)
destination_client.add_alias('emo_cowboy', 'sad_cowboy')
```

Comment (gmkohler) [reply emoji/emoji_transfer_service.py]: If we only specify `'emo_cowboy'` in the input, I don't see the need to add both names.

**Pull request merged**